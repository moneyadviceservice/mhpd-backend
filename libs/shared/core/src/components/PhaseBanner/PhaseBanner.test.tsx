import React from 'react';
import { render, screen } from '@testing-library/react';
import { PhaseBanner, PhaseType } from '.';

jest.mock('next/router', () => ({
  useRouter: jest.fn().mockReturnValue({
    query: {},
  }),
}));

describe('Phase banner component', () => {
  it('renders beta correctly', () => {
    render(<PhaseBanner />);
    const betaBanner = screen.getByTestId('phase-banner');
    expect(betaBanner).toMatchSnapshot();
  });

  it('renders alpha correctly', () => {
    render(<PhaseBanner phase={PhaseType.ALPHA} />);
    const betaBanner = screen.getByTestId('phase-banner');
    expect(betaBanner).toMatchSnapshot();
  });
});
