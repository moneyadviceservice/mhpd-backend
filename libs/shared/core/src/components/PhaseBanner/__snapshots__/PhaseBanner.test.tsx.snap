// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Phase banner component renders alpha correctly 1`] = `
<div
  class="mt-3 pb-4 border-b-1 border-slate-400 flex items-center leading-[23px]"
  data-testid="phase-banner"
>
  <span
    class="mr-3 py-[5px] px-[8px] uppercase text-sm font-bold bg-pink-600 text-white"
  >
    alpha
  </span>
  <span>
    This is a new service - your
     
    feedback
     
    will help us to improve it.
  </span>
</div>
`;

exports[`Phase banner component renders beta correctly 1`] = `
<div
  class="mt-3 pb-4 border-b-1 border-slate-400 flex items-center leading-[23px]"
  data-testid="phase-banner"
>
  <span
    class="mr-3 py-[5px] px-[8px] uppercase text-sm font-bold bg-pink-600 text-white"
  >
    beta
  </span>
  <span>
    This is a new service - your
     
    feedback
     
    will help us to improve it.
  </span>
</div>
`;
